src/Compiler/Microsoft.AspNetCore.Razor.Language/test/Legacy/CSharpTokenizerLiteralTest.cs

```diff
@@ -16,137 +16,171 @@ public class CSharpTokenizerLiteralTest : CSharpTokenizerTestBase
     }
 
     [Fact]
     public void Trailing_Letter_Is_Not_Part_Of_Integer_Literal_If_Not_Type_Sufix()
     {
-        TestTokenizer("42a", SyntaxFactory.Token(SyntaxKind.IntegerLiteral, "42"), IgnoreRemaining);
+        TestTokenizer("42a", SyntaxFactory.Token(SyntaxKind.NumericLiteral, "42"), IgnoreRemaining);
     }
 
     [Fact]
     public void Simple_Hex_Literal_Is_Recognized()
     {
-        TestSingleToken("0x0123456789ABCDEF", SyntaxKind.IntegerLiteral);
+        TestSingleToken("0x0123456789ABCDEF", SyntaxKind.NumericLiteral);
     }
 
     [Fact]
     public void Integer_Type_Suffix_Is_Recognized_In_Hex_Literal()
     {
-        TestSingleToken("0xDEADBEEFU", SyntaxKind.IntegerLiteral);
-        TestSingleToken("0xDEADBEEFu", SyntaxKind.IntegerLiteral);
+        TestSingleToken("0xDEADBEEFU", SyntaxKind.NumericLiteral);
+        TestSingleToken("0xDEADBEEFu", SyntaxKind.NumericLiteral);
 
-        TestSingleToken("0xDEADBEEFL", SyntaxKind.IntegerLiteral);
-        TestSingleToken("0xDEADBEEFl", SyntaxKind.IntegerLiteral);
+        TestSingleToken("0xDEADBEEFL", SyntaxKind.NumericLiteral);
+        TestSingleToken("0xDEADBEEFl", SyntaxKind.NumericLiteral);
 
-        TestSingleToken("0xDEADBEEFUL", SyntaxKind.IntegerLiteral);
-        TestSingleToken("0xDEADBEEFUl", SyntaxKind.IntegerLiteral);
+        TestSingleToken("0xDEADBEEFUL", SyntaxKind.NumericLiteral);
+        TestSingleToken("0xDEADBEEFUl", SyntaxKind.NumericLiteral);
 
-        TestSingleToken("0xDEADBEEFuL", SyntaxKind.IntegerLiteral);
-        TestSingleToken("0xDEADBEEFul", SyntaxKind.IntegerLiteral);
+        TestSingleToken("0xDEADBEEFuL", SyntaxKind.NumericLiteral);
+        TestSingleToken("0xDEADBEEFul", SyntaxKind.NumericLiteral);
 
-        TestSingleToken("0xDEADBEEFLU", SyntaxKind.IntegerLiteral);
-        TestSingleToken("0xDEADBEEFLu", SyntaxKind.IntegerLiteral);
+        TestSingleToken("0xDEADBEEFLU", SyntaxKind.NumericLiteral);
+        TestSingleToken("0xDEADBEEFLu", SyntaxKind.NumericLiteral);
 
-        TestSingleToken("0xDEADBEEFlU", SyntaxKind.IntegerLiteral);
-        TestSingleToken("0xDEADBEEFlu", SyntaxKind.IntegerLiteral);
+        TestSingleToken("0xDEADBEEFlU", SyntaxKind.NumericLiteral);
+        TestSingleToken("0xDEADBEEFlu", SyntaxKind.NumericLiteral);
     }
 
     [Fact]
-    public void Trailing_Letter_Is_Not_Part_Of_Hex_Literal_If_Not_Type_Sufix()
+    public void Trailing_Letter_Is_Not_Part_Of_Hex_Literal_If_Not_Type_Suffix()
     {
-        TestTokenizer("0xDEADBEEFz", SyntaxFactory.Token(SyntaxKind.IntegerLiteral, "0xDEADBEEF"), IgnoreRemaining);
+        TestTokenizer("0xDEADBEEFz", SyntaxFactory.Token(SyntaxKind.NumericLiteral, "0xDEADBEEF"), IgnoreRemaining);
+    }
+
+    [Fact]
+    public void Binary_Literal_Is_Recognized()
+    {
+        TestSingleToken("0b01010101", SyntaxKind.NumericLiteral);
+    }
+
+    [Fact]
+    public void Integer_Type_Suffix_Is_Recognized_In_Binary_Literal()
```

Comment:
Did we support binary literals before? Is this a new Razor language feature? :smile:

Reply:
It's a new feature indeed.